### PR TITLE
Replace MCAuth with RK_01 MinecraftAuth

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**

### CODE REVIEW

File: auth/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * SkinsRestorer
```
Comment (Camotoy) [inline auth/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: uh

Comment (AlexProgrammerDE) [reply auth/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: lmao 🤣

**Comment by Konicai** [comment]
Let's not just copy and paste GNU licensed code into an MIT repo and strip the license...

**Comment by AlexProgrammerDE** [comment]
I made that code, you have explicit permission to use it.

**Comment by Konicai** [comment]
ah yeah i remember now, sorry

thanks :P

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Information about a user profile.
+ */
+public class GameProfile {
+    private static final String[] WHITELISTED_DOMAINS = {".minecraft.net", ".mojang.com"};
```
Comment (rtm516) [inline protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java]: Should we not include the block list like the mojang authlib?

Currently blocks:
- bugs.mojang.com
- education.minecraft.net
- feedback.minecraft.net

Seen in `com.mojang.authlib.yggdrasil.TextureUrlChecker`

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+        return false;
+    }
+
+    private final UUID id;
+    private final String name;
+
+    private List<Property> properties;
+    private Map<TextureType, Texture> textures;
+    private boolean texturesVerified;
```
Comment (rtm516) [inline protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:46]: These should be above the static init block

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * Utilities for making HTTP requests.
+ */
+public class HTTPUtils {
+    private static final Gson GSON = new GsonBuilder()
+            .registerTypeAdapter(UUID.class, new UndashedUUIDAdapter())
+            .create();
+
+    private HTTPUtils() {
+    }
```
Comment (rtm516) [inline protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java:32]: Remove the empty constructor

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java (hunk 4 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java:32]: No, other utility classes around the codebase also follow this standard pattern:
image

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java (hunk 3 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:46]: No, other classes like ServerListener have the non-static fields below the static block.

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java (hunk 4 above)
Comment (rtm516) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/HTTPUtils.java:32]: Ah ok, something to likely look at in future then

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java (hunk 3 above)
Comment (rtm516) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/GameProfile.java:46]: Ah ok, something to likely look at in future then

File: example/src/main/java/org/geysermc/mcprotocollib/protocol/example/MinecraftProtocolTest.java [hunk 5]
```diff
@@ -177,19 +176,21 @@ private static void status() {
...
-                return;
+                fullJavaSession = MinecraftAuth.JAVA_CREDENTIALS_LOGIN.getFromInput(
+                        MinecraftAuth.createHttpClient(),
+                        new StepCredentialsMsaCode.MsaCredentials(USERNAME, PASSWORD));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
             }
+
+            var mcProfile = fullJavaSession.getMcProfile();
+            var mcToken = mcProfile.getMcToken();
```
Comment (Kas-tle) [inline example/src/main/java/org/geysermc/mcprotocollib/protocol/example/MinecraftProtocolTest.java]: We do not use var anywhere else in the code base.

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        if (noDashes == null) {
+            return null;
+        }
+
+        StringBuilder idBuff = new StringBuilder(noDashes);
+        idBuff.insert(20, '-');
+        idBuff.insert(16, '-');
+        idBuff.insert(12, '-');
+        idBuff.insert(8, '-');
+        return UUID.fromString(idBuff.toString());
```
Comment (Kas-tle) [inline protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: This seems simpler:
```java
return new UUID(
    java.util.HexFormat.fromHexDigitsToLong(noDashes, 0, 16),
    java.util.HexFormat.fromHexDigitsToLong(noDashes, 16, 32)
);
```

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -4,12 +4,13 @@ metadata.format.version = "1.1"
...
 adventure = "4.15.0"
 cloudburstnbt = "3.0.0.Final"
-mcauthlib = "e5b0bcc"
 slf4j = "2.0.9"
 math = "2.0"
 fastutil-maps = "8.5.3"
 netty = "4.1.103.Final"
 netty-io_uring = "0.0.24.Final"
+gson = "2.10.1"
+minecraftauth = "4.0.1"
```
Comment (Kas-tle) [inline gradle/libs.versions.toml]: Update to latest versions.

**Review by Kas-tle — APPROVED** [review]

**Comment by Kas-tle** [comment]
Also please resolve merge conflict in ClientboundStatusResponsePacket.

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java (hunk 6 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: That seems slower to me

Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: I'll do a microbenchmark

File: example/src/main/java/org/geysermc/mcprotocollib/protocol/example/MinecraftProtocolTest.java (hunk 5 above)
Comment (AlexProgrammerDE) [reply example/src/main/java/org/geysermc/mcprotocollib/protocol/example/MinecraftProtocolTest.java]: Ah yeah I see, took this from my SoulFire codebase. I exclusively use var for a few months and i copy-pasted that.

**Comment by Kas-tle** [comment]
Will merge tomorrow if there are no objections.

File: protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java (hunk 6 above)
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java]: image
Alright seems like your code is 3.6x faster than mine.

**Review by rtm516 — APPROVED** [review]

**Pull request merged**